packages/components/src/components/common/inputtag/inputBadge.tsx

```diff
@@ -0,0 +1,79 @@
+import { forwardRef, type ElementRef } from 'react';
+import { useField } from '../field/field';
+import { IvyIcon } from '../icon/icon';
+import { inputBadge, inputBadgeIcon, inputBadgeLine, inputBadgeOutput } from './inputBadge.css';
+
+import type { IvyIcons } from '@axonivy/ui-icons';
+
+export type InputBadgeProps = {
+  value: string;
+  badgeRegex: BadgeRegex[];
+};
+
+type BadgeRegex = { delimiter: { start: string; end: string }; icon: IvyIcons; badgeTextGen: (text: string) => string };
+
+export type BadgeTextProps = { icon: IvyIcons | undefined; text: string };
```

Comment:
if you want an optional parameter it is better to use the `?` operator. 
Because if you do it with `| undefined` you have to pass an argument, at least `undefined`. e.g `{icon: undefined, text: 'hi'}`
But if you do it with the `?`, `{icon?: IvyIcons, text: string}`, then you don't have to pass anything for this attribute: `{text: 'hi'}`